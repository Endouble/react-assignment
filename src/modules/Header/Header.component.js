import React from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '../Button';
import Flash from '../Button/Flash.component';

import { colors } from '../../styles/vars';
import { boldBorders } from '../../styles/mixins';

const Header = () => (
    <StyledHeader>
        <div className="pokeHeader__flash">
            <Flash />
        </div>
        <div className="pokeHeader__buttons">
            <Button sizeIndex={4} backColor={colors.red} />
            <Button sizeIndex={4} backColor={colors.yellow} />
            <Button sizeIndex={4} backColor={colors.green} />
        </div>
    </StyledHeader>
);

/*
    Header Styles
*/
const StyledHeader = styled.header`
    background: ${colors.magenta};
    border-bottom: 8px solid ${colors.darkPurple};
    ${boldBorders()};
    display: flex;

    .pokeHeader__flash {
        padding: 15px;
        text-align: center;
    }

    .pokeHeader__buttons {
        overflow: hidden;
        display: flex;
        padding-top: 8px;

        > div {
            margin-right: 15px;
        }
    }

`;

/*
    Header propTypes
*/
Header.propTypes = {
    // children: PropTypes.node,
};
Header.defaultProps = {
    // children: null,
};

export default Header;
