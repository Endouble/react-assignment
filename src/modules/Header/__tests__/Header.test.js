import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../';

configure({ adapter: new Adapter() });

describe('Header', () => {
    it('renders correctly', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });
    it('Has Flash component', () => {
        const component = shallow(<Header />);
        const hasFlash = component.find('.pokeHeader__flash').exists();
        expect(hasFlash).toEqual(true);
    });
    it('Has Buttons wrapper', () => {
        const component = shallow(<Header />);
        const hasButtons = component.find('.pokeHeader__buttons').exists();
        expect(hasButtons).toEqual(true);
    });
});
