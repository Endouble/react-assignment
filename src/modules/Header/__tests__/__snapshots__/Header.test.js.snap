// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<styled.header>
  <div
    className="pokeHeader__flash"
  >
    <FlashData />
  </div>
  <div
    className="pokeHeader__buttons"
  >
    <Button
      backColor="#fd1b55"
      className=""
      isPlain={true}
      sizeIndex={4}
    />
    <Button
      backColor="#f7ee60"
      className=""
      isPlain={true}
      sizeIndex={4}
    />
    <Button
      backColor="#30f454"
      className=""
      isPlain={true}
      sizeIndex={4}
    />
  </div>
</styled.header>
`;
